src/pages/Settings.tsx

```diff
@@ -0,0 +1,151 @@
+import React, { useContext, useEffect, useRef, useState } from "react"
+import { View } from "react-native"
+import { SettingsStackScreen, useNavigation } from "navigation/NavigationTypes"
+import { SettingsScroll } from "components/Settings"
+import { Divider } from "components/Divider"
+import { SettingTile } from "components/Settings"
+import { settingsIcons } from "assets/settings"
+import { UserDetailsTile } from "components/Settings"
+import { Setting } from "components/Settings"
+import { ModalCustomSettings } from "components/Settings"
+import { CourseTile } from "components/Settings"
+import { SelectTile } from "components/Settings"
+import { AppContext } from "../state/AppContext"
+import { RadioButtonGroup } from "components/Settings"
+import AsyncStorage from "@react-native-async-storage/async-storage"
+import { UserAnonymousTile } from "components/Settings"
+
+const themes: string[] = ["Predefinito", "Scuro", "Chiaro"]
+const themesToSave: string[] = ["predefined", "dark", "light"]
+
+/**
+ * Settings Page
+ */
+export const SettingsPage: SettingsStackScreen<"Settings"> = props => {
+    const user = props.route.params.user
+
+    const context = useContext(AppContext)
+
+    const settings = context.settings
+    const setSettings = context.setSettings
+
+    //App state theme
+    const theme = settings.theme
+    //Settings setter
+
+    //RadioButtonGroup theme state and setter
+    const [selectedTheme, setSelectedTheme] = useState(theme)
+
+    //RadioButtonGroup carriere state and setter
+    const [carriera, setCarriera] = useState(
+        user.carriere[0].matricola.toString()
+    )
+
+    //for testing logged in/out view
+    const [logged, setLogged] = useState(false)
+
+    const [isModalVisible, setModalVisible] = useState(false)
+
+    //tracking first render
+    const firstRender = useRef(true)
+
+    const { navigate } = useNavigation()
+
+    //Update storage as a side effect of settings change
+    useEffect(() => {
+        // ? skip on first render, is it worth it ?
+        if (firstRender.current) {
+            firstRender.current = false
+        } else {
+            AsyncStorage.setItem("settings", JSON.stringify(settings)).catch(
+                err => console.log(err)
+            )
+            console.log("Set theme " + theme)
+        }
+    }, [theme])
```

Comment:
Lo `useEffect` per scrivere su disco dovrebbe essere in App.tsx, a livello di astrazione deve essere un side effect della modifica dello stato settings, quindi quando questo varia viene scritto su disco

in App.tsx mi immagino una cosa banale tipo:
```ts
    useEffect(() => {
        void AsyncStorage.setItem("settings", JSON.stringify(settings))
    }, [settings])
```

E in generale, in Settings.tsx o in qualsiasi campo che modifica un impostazione sarà semplicemente
```tsx
const { settings, setSettings } = useContext(SettingsContext)

...

<ComponenteMagico
    value={settings.campoMagico}
    onChange(newValue => {
        setSettings({ ...settings, campoMagico: newValue })
    })
/>
```
Molto più streamlined e ordinato, ogni cosa sta al suo posto senza avere side effect buttati a cazzo chissà dove

Reply:
Okay va bene. Lascio comunque la logica per skippare il primo render ok? Altrimenti si fa un set inutile dell'oggetto appena creato, anche se viene sovrascritto immediatamente dopo da quello parsed preso dallo storage. In realtà si potrebbe anche skippare il secondo setItem visto che salva ciò che viene preso dallo storage (e dato che si trova già nello storage non serve riscriverlo) ma forse non vale la pena aggiungere questa condizione in più e complica troppo. Comunque fra poco faccio il commit cosi poi vedi.